shipyard-controller/common/filereader_test.go

```diff
@@ -0,0 +1,105 @@
+package common
+
+import (
+	"testing"
+	"testing/fstest"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_FileReader(t *testing.T) {
+	denyListTestFileName := "test-file"
+	tests := []struct {
+		provider FileReader
+		result   []string
+	}{
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{
+					denyListTestFileName: {Data: []byte("some\nstring")},
+				},
+			},
+			result: []string{"some", "string"},
+		},
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{
+					denyListTestFileName: {Data: []byte("some\nstring\n\n\n")},
+				},
+			},
+			result: []string{"some", "string"},
+		},
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{
+					denyListTestFileName: {Data: []byte("some\n\n\nstring\n\n\n")},
+				},
+			},
+			result: []string{"some", "string"},
+		},
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{
+					denyListTestFileName: {Data: []byte("some")},
+				},
+			},
+			result: []string{"some"},
+		},
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{
+					denyListTestFileName: {Data: []byte("")},
+				},
+			},
+			result: []string{},
+		},
+		{
+			provider: &fileReader{
+				FileSystem: fstest.MapFS{},
+			},
+			result: []string{},
+		},
+	}
```

Comment:
would it be easy to add a test for testing a file error?

Reply:
we are already testing it, it is the last test